src/olympia/files/tests/test_utils.py

```diff
@@ -1216,65 +1217,122 @@ def test_raises_error_for_absolute_path(self):
             utils.SafeTar.open(filename)
 
 
-class TestArchiveMemberValidator(TestCase):
-    # We cannot easily test `archive_member_validator` so let's test
-    # `_validate_archive_member_name_and_size` instead.
+class TestArchiveMemberValidatorZip(TestCase):
+    def _fake_archive_member(self, filename, filesize):
+        info = zipfile.ZipInfo(filename)
+        info.file_size = filesize
+        return info
 
     def test_raises_when_filename_is_none(self):
+        info = self._fake_archive_member('', 123)
+        info.filename = None
         with pytest.raises(utils.InvalidArchiveFile):
-            utils._validate_archive_member_name_and_size(None, 123)
+            utils.archive_member_validator(info)
+
+    def test_unsupported_compression_method(self):
+        info = self._fake_archive_member('foo', 123)
+        info.compress_type == zipfile.ZIP_DEFLATED
+        # ZIP_DEFLATED works.
+        utils.archive_member_validator(info)
+        # ZIP_STORED works.
+        info.compress_type = zipfile.ZIP_DEFLATED
+        utils.archive_member_validator(info)
+
+        # The rest should raise an error. Test a few known ones out there:
+        ZIP_DEFLATED64 = 9
+        ZIP_IMPLODED = 10
+        ZIP_BZIP2 = 12
+        ZIP_LZMA = 14
+        ZIP_ZSTANDARD = 93
+        ZIP_XZ = 95
+        ZIP_PPMD = 98
+        for compress_type in (
+            ZIP_DEFLATED64,
+            ZIP_IMPLODED,
+            ZIP_BZIP2,
+            ZIP_LZMA,
+            ZIP_ZSTANDARD,
+            ZIP_XZ,
+            ZIP_PPMD,
+        ):
+            info.compress_type = zipfile.ZIP_LZMA
```

Comment:
You're always setting `compress_type` to `zipfile.ZIP_LZMA`, as opposed to using `compress_type` from the loop.

Reply:
doh, I added the loop late, didn't re-read it. I'll fix it.

Reply:
ead5ccc56acda3b3a22a6b845d9e23a9c1fe91c2